src/cgutils.cpp

```diff
@@ -1075,6 +1075,9 @@ static Value *emit_typeptr_addr(Value *p)
     return emit_nthptr_addr(p, -offset);
 }
 
+static Value *boxed(const jl_cgval_t &v, jl_codectx_t *ctx, bool gcooted=true);
+static Value *boxed(const jl_cgval_t &v, jl_codectx_t *ctx, jl_value_t* type) = delete; // C++11 (temporary to prevent rebase error)
```

Comment:
correct. C++ will silently convert from a `jl_value_t*` to `bool` during dispatch, potentially leading to unhappiness. this makes it a fast error, but won't compile on pre-C++11. if someone needs a pre-C++11 compiler and complains about this, we can either remove the `= delete` (turning misuse into a linker error), or the whole line (turning misuse into silent corruption).